rest/src/main/java/org/jboss/pnc/rest/provider/BuildConfigurationProvider.java

```diff
@@ -255,10 +265,43 @@ public BuildConfigurationAuditedRest getRevision(Integer id, Integer rev) {
         return new BuildConfigurationAuditedRest (auditedBuildConfig);
     }
 
+    public Response getLatestBuildRecord(Integer configId) {
+        BuildConfiguration buildConfiguration = buildConfigurationRepository.queryById(configId);
+        if (buildConfiguration == null) {
+            return Response.status(Response.Status.NOT_FOUND).entity("No build configuration exists with id: " + configId).build();
+        }
+        PageInfo pageInfo = this.pageInfoProducer.getPageInfo(0, 1);
+        SortInfo sortInfo = this.sortInfoProducer.getSortInfo(SortingDirection.DESC, "endTime");
+        List<BuildRecord> buildRecords = buildRecordRepository.queryWithPredicates(pageInfo, sortInfo, withBuildConfigurationId(configId));
```

Comment:
You could also use this to make it even simpler: https://github.com/project-ncl/pnc/blob/master/spi/src/main/java/org/jboss/pnc/spi/datastore/repositories/api/ReadOnlyRepository.java#L12